dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java

```diff
@@ -0,0 +1,97 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.http;
+
+import com.google.inject.Inject;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpUtil;
+
+import static io.netty.handler.codec.http.HttpHeaderNames.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
+import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import alluxio.client.file.cache.CacheManager;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HttpServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+    private final PagedService mPagedService;
+
+    public HttpServerHandler(PagedService pagedService) {
+        mPagedService = pagedService;
+    }
+
+    @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) {
+        ctx.flush();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
+        if (msg instanceof HttpRequest) {
+            HttpRequest req = (HttpRequest) msg;
+            String requestUri = req.uri();
+            // parse the request uri to get the parameters
+            requestUri = requestUri.substring(requestUri.indexOf("?") + 1);
+            String[] params = requestUri.split("&");
+            Map<String, String> parametersMap = new HashMap<>();
+            for (String param : params) {
+                String[] keyValue = param.split("=");
+                parametersMap.put(keyValue[0], keyValue[1]);
+            }
+            String fileId = parametersMap.get("fileId");
+            long pageIndex = Long.parseLong(parametersMap.get("pageIndex"));
+
+            boolean keepAlive = HttpUtil.isKeepAlive(req);
+            ByteBuf byteBuf = mPagedService.getPage(fileId, pageIndex, ctx.channel());
+            FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), OK,
```

Comment:
just curious would we still have zero-copy if we put ByteBuf here instead of DataFileChannel/FileRegion?

Reply:
If we don't use FileRegion, it is without zero-copy. This just want to implement the get page RESTful API first. And then replace it with FileRegion.